### DISCUSSION TITLE
Completely support of gzip compression in mongoose

### DISCUSSION THREAD

**Discussion by ghost** [cesanta/mongoose#1926] [Feature requests]
Hi again
mongoose already handles the gzip compression in mg_http_serve_file(). but that function is not check the request header "Accept-Encoding:gzip", Also FILE.gz does not have priority over original File.
thanks for 
[mongoose-completely-support-file-gz.patch.gz](https://github.com/cesanta/mongoose/files/10251000/mongoose-completely-support-file-gz.patch.gz)

**Comment by scaprile** [comment]
> Hi again mongoose already handles the gzip compression in mg_http_serve_file(). but that function is not check the request header "Accept-Encoding:gzip", Also FILE.gz does not have priority over original File. 

No, and no, the comment explains the reason (this is embedded) and there is no priority (as the docs explain), if you want gzipped then don't put uncompressed files
